### DISCUSSION TITLE
Really nice!

### DISCUSSION THREAD

**Discussion by mitchsf** [gemi254/ConfigAssist-ESP32-ESP8266#1] [General]
This is exactly what I need. My application has around 70 parameters to set. I use mostly dropdowns, a few ranges, and a couple text fields. I'll replace my crude attempt to do this, with ConfigAssist.

I would have preferred that it was designed as a library with many of the functions in the .ino incorporated, but this will work as is. The application configuration dictionary will of course, be moved to a separate, .h file.

Thanks for this.

**Comment by gemi254** [comment]
Thanks
Check..
#define MAX_PARAMS 50                // Maximum parameters to handle

if your variables exceed MAX_PARAMS increase this value in class header configAssist.h

"designed as a library" meaning with a .h and a .cpp ?

I use visual studio code and it is more easy but it easy to implement..

**Reply by mitchsf** [reply]
I'm using the latest Arduino IDE and ESP32 core. A library isn't really necessary, that's just what I'm used to working with.

A couple things I can think of that I will need - A flag that will force the json and .ini files to be deleted. With an application firmware update, parameters may be added, changed, etc., so these files can be recreated. It will also make testing and development easier.

Also, I have one dropdown that will have between one and twenty parameters passed to it as options. No fixed text strings. I'm working on a way to do that.

I just noticed that the demo is using 87% of program storage space on the ESP32 Pico, which is surprising. 

Thanks again.

**Comment by gemi254** [comment]
Finaly converted ConfigAssist to an arduino library..

If you want to remove .ini files and recreate them press the ''Defaults'' button on /cfg page
or
//Uncomment to remove ini file and re-built it fron dictionary
//conf.deleteConfig();

To make drop-down dynamic see how can be done (dynamic list of wifi scan)

configAssistPMem.h

async function getWifiScan(){

**Reply by mitchsf** [reply]
Thanks!
